src/d3a/models/strategy/external_strategies/__init__.py

```diff
@@ -444,38 +443,102 @@ def _set_energy_forecast(self, payload):
                 IncomingRequest("set_energy_forecast", arguments,
                                 energy_forecast_response_channel))
 
-    def _set_energy_forecast_impl(self, arguments, response_channel):
+    def set_energy_measurement(self, payload: Dict) -> None:
+        """Callback of set_energy_measurement redis endpoint."""
+        transaction_id = self._get_transaction_id(payload)
+        energy_measurement_response_channel = \
+            f"{self.channel_prefix}/response/set_energy_measurement"
         try:
-            assert arguments["energy_forecast"] >= 0.0
-            self.energy_forecast_buffer_Wh = arguments["energy_forecast"]
+            arguments = json.loads(payload["data"])
+            assert set(arguments.keys()) == {"energy_measurement", "transaction_id"}
+        except Exception:
+            logging.exception(
+                f"Incorrect _set_energy_measurement request. Payload: {payload}.")
+            self.redis.publish_json(
+                energy_measurement_response_channel,
+                {"command": "set_energy_measurement",
+                 "error": "Incorrect _set_energy_measurement request. "
+                          "Available parameters: (energy_measurement).",
+                 "transaction_id": transaction_id})
+        else:
+            self.pending_requests.append(
+                IncomingRequest("set_energy_measurement", arguments,
+                                energy_measurement_response_channel))
+
+    @staticmethod
+    def _validate_values_positive_in_profile(profile: Dict) -> None:
+        """Validate whether all values are positive in a profile."""
+        for time_str, energy in profile.items():
+            if energy < 0.0:
+                raise ValueError(f"Energy is not positive for time stamp {time_str}.")
+
+    def set_energy_forecast_impl(self, arguments: Dict, response_channel: str) -> None:
+        """
+        Digest command from buffer and perform set_energy_forecast.
+        Args:
+            arguments: Dictionary containing "energy_forecast" that is a dict containing a profile
+                key: time string, value: energy in kWh
```

Comment:
We might be obliged to also adapt the device measurement subscriber to support this, I'll do it in the context of D3ASIM-3549

Reply:
yes, you are right, I wanted to mention this in the PR description, but forgot.

Reply:
> I'll do it in the context of D3ASIM-3549

Thanks a lot!